### PR TITLE
Unit tests for groupapprove auto approve.

### PR DESCRIPTION
**Pull request by jcbsmpsn**

### CODE REVIEW

File: pkg/controller/certificates/groupapprove_test.go [hunk 1]
```diff
@@ -69,3 +75,140 @@ func TestHasKubeletUsages(t *testing.T) {
...
+		csr:                        workingCertificateSigningRequest,
+		expectError:                false,
+		expectedApproved:           true,
+		expectedDenied:             false,
+		expectedNumberOfConditions: 1,
+	}}
+	successTestCase := testCases[0]
+
+	tc := successTestCase
+	tc.description = "No approval group configured"
```
Comment (deads2k) [inline pkg/controller/certificates/groupapprove_test.go]: Why construct the list of tests this way instead of during intialization like other tests?

**Comment by deads2k** [comment]
Just the question about construction.  It looks odd.

File: pkg/controller/certificates/groupapprove_test.go (hunk 1 above)
Comment (jcbsmpsn) [inline pkg/controller/certificates/groupapprove_test.go]: The base is a success test case. Then I clone it and take away individual elements to make it a failing test case. I would have had to manually duplicate the working test case many times to use the usual initialization procedure.

**Pull request closed without merging**